Core/Utilities.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.IO;
+
+namespace CKAN
+{
+    public static class Utilities
+    {
+        /// <summary>
+        /// Copies a directory and optionally its subdirectories.
+        /// </summary>
+        /// <param name="sourceDirPath">Source directory path.</param>
+        /// <param name="destDirPath">Destination directory path.</param>
+        /// <param name="copySubDirs">Copy sub dirs recursively if set to <c>true</c>.</param>
+        public static void CopyDirectory(string sourceDirPath, string destDirPath, bool copySubDirs)
```

Comment:
Could/should we do this within a Transaction? Seems like this is as likely to fail as installing a mod would be, and it would be nice to be able to auto clean it up if that happens.

Reply:
Does that work the way I did it now?